### PR TITLE
Add base files

### PR DESCRIPTION
**Pull request by romkatsu**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | Add base files

### CODE REVIEW

**Comment by BoShurik** [comment]
Does template need to contain example entities? Maybe create separate `yii-demo-api` for this?

File: README.md [hunk 1]
```diff
@@ -2,46 +2,60 @@
     <a href="https://github.com/yiisoft" target="_blank">
         <img src="https://github.com/yiisoft.png" height="100px">
     </a>
-    <h1 align="center">Yii _____</h1>
+    <h1 align="center">Yii api template</h1>
```
Comment (samdark) [inline README.md:5]:
```suggestion
    <h1 align="center">Yii API template</h1>
```

File: README.md [hunk 2]
```diff
@@ -2,46 +2,60 @@
...
-[![static analysis](https://github.com/yiisoft/_____/workflows/static%20analysis/badge.svg)](https://github.com/yiisoft/_____/actions?query=workflow%3A%22static+analysis%22)
+[![Latest Stable Version](https://poser.pugx.org/yiisoft/app-api/v/stable.png)](https://packagist.org/packages/yiisoft/app-api)
+[![Total Downloads](https://poser.pugx.org/yiisoft/app-api/downloads.png)](https://packagist.org/packages/yiisoft/app-api)
+[![Build status](https://github.com/yiisoft/app-api/workflows/build/badge.svg)](https://github.com/yiisoft/app-api/actions?query=workflow%3Abuild)
+[![Scrutinizer Code Quality](https://scrutinizer-ci.com/g/yiisoft/app-api/badges/quality-score.png?b=master)](https://scrutinizer-ci.com/g/yiisoft/app-api/?branch=master)
+[![static analysis](https://github.com/yiisoft/app-api/workflows/static%20analysis/badge.svg)](https://github.com/yiisoft/app-api/actions?query=workflow%3A%22static+analysis%22)
 
-The package ...
+Yii api application template for Yii 3
```
Comment (samdark) [inline README.md:15]:
```suggestion
Yii API application template for Yii 3
```

File: README.md [hunk 3]
```diff
@@ -2,46 +2,60 @@
...
 
 ## Installation
 
-The package could be installed with composer:
+Install docker:
+```
+docker-compose up -d
+```
+
+Enter in to the container:
```
Comment (samdark) [inline README.md:24]:
```suggestion
Enter into the container:
```

File: README.md [hunk 4]
```diff
@@ -2,46 +2,60 @@
...
+| POST /blog/      | yes           | create blog record   |
+| PUT  /blog/[id]  | yes           | update blog record   |
+| GET  /users/     | yes           |  get users           |
+| GET  /users/[id] | yes           |  get user            |
+| POST /auth/      | no            |  auth                |
 
-```php
-./vendor/bin/phpunit
-```
+Authorization is performed via the header X-Api-Key
```
Comment (samdark) [inline README.md:49]:
```suggestion
Authorization is performed via the `X-Api-Key` header.
```

File: composer.json [hunk 5]
```diff
@@ -1,45 +1,118 @@
 {
-    "name": "yiisoft/_____",
-    "type": "library",
-    "description": "_____",
+    "name": "yiisoft/app-api",
+    "type": "project",
+    "description": "Yii Framework Project Api Demo",
```
Comment (samdark) [inline composer.json:4]:
```suggestion
    "description": "Yii Framework API project template",
```

File: composer.json [hunk 6]
```diff
@@ -1,45 +1,118 @@
...
     "keywords": [
-        "_____"
+        "yii3",
+        "app",
+        "api"
     ],
-    "homepage": "https://www.yiiframework.com/",
+    "homepage": "https://github.com/romkatsu/yii-api-demo",
```
Comment (samdark) [inline composer.json:10]:
```suggestion
    "homepage": "https://github.com/yiisoft/app-api/",
```

File: config/common.php [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+$timer->start('overall');
+
+return [
+    ContainerInterface::class => static function (ContainerInterface $container) {
+        return $container;
+    },
+    RouteCollectorInterface::class => Group::create(),
+    UrlMatcherInterface::class => new AppRouterFactory(),
+    UrlGeneratorInterface::class => UrlGenerator::class,
+    Timer::class => $timer,
```
Comment (samdark) [inline config/common.php:26]: It's not used anywhere, isn't it?

**Comment by samdark** [comment]
@BoShurik comment is valid. If it is a template, there should not be too much code that is to be deleted. Tokens, user auth etc. is OK but full featured blog example is better to be done as part of the guide.

For now, though, I'd leave it as is.

File: config/common.php (hunk 7 above)
Comment (romkatsu) [reply config/common.php]: removed

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**